metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/security/AuthorizationServerHelper.java

```diff
@@ -0,0 +1,129 @@
+            mockRequest.addHeader("Authorization", "Bearer " + token.getValue());
+            return mockRequest;
+        };
+    }
+
+    private OAuth2AccessToken createAccessToken(final String clientId) {
+        ClientDetails client = clientDetailsService.loadClientByClientId(clientId);
+        Map<String, Object> clientNameMap = client.getAdditionalInformation();
+
+        OAuth2Authentication auth = jwtAccessTokenCustomizer(mapper).extractAuthentication(clientNameMap);
+        OAuth2AccessToken auth2AccessToken = tokenservice.createAccessToken(auth);
+        return auth2AccessToken;
+    }
+
+    @Override
+    public void configure(final ClientDetailsServiceConfigurer clients) throws Exception {
+        Map<String, Object> testuserMap = getClientInfo("testuser");
+        Map<String, Object> testoperatorMap = getClientInfo("testoperator");
+        clients.inMemory()
+                .withClient("testoperator")
+                .authorities("ROLE_SERVICE_OPERATOR")
+                .additionalInformation(testoperatorMap)
+                .and()
+                .withClient("testuser")
+                .additionalInformation(testuserMap);
+    }
+
+    @Bean
+    public JwtAccessTokenCustomizer jwtAccessTokenCustomizer(ObjectMapper mapper) {
+        return new JwtAccessTokenCustomizer(mapper);
+    }
+
+    private static Map<String, Object> getClientInfo(String clientId) {
+        Map<String, Object> clientNameMap = new LinkedHashMap<>();
+        clientNameMap.put("jti", "7ac94e65-119e-471e-abb9-4e1fb5cc79d2");
+        clientNameMap.put("exp", 1561536742L);
+        clientNameMap.put("nbf", 0);
+        clientNameMap.put("iat", 1561536442);
+        clientNameMap.put("iss", "http://localhost:8085/ampt2d/auth/realms/securemetadata");
+        clientNameMap.put("aud", "secure-client");
+        clientNameMap.put("sub", "77dfbb1a-f998-4477-8276-f7a3ef5893b0");
+        clientNameMap.put("typ", "Bearer");
+        clientNameMap.put("azp", "secure-client");
+        clientNameMap.put("auth_time", 1561536442);
+        clientNameMap.put("session_state", "4a75616f-810c-40ed-beb7-875867b0accc");
+        clientNameMap.put("acr", "1");
+        clientNameMap.put("allowed-origins", new ArrayList<>());
+
+        ArrayList<String> rolesValueRealmList = new ArrayList<>(Arrays.asList("offline_access", "uma_authorization"));
+        Map<String, Object> rolesMap = new LinkedHashMap<>();
+        rolesMap.put("roles", rolesValueRealmList);
+        clientNameMap.put("realm_access", rolesMap);
+
+        ArrayList<String> rolesValueList = new ArrayList<>(Arrays.asList("manage_account", "manage_account-links", "view-profile"));
+        Map<String, Object> accountMap = new LinkedHashMap<>();
+        Map<String, Object> resourceAccessMap = new LinkedHashMap<>();
+        accountMap.put("roles", rolesValueList);
+        resourceAccessMap.put("account", accountMap);
+        if (clientId.equals("testoperator")) {
```

Comment:
Oh, one more thing: it's not very reliable that the choice of role depends on a `clientId`; conceptually it is another parameter. I think it's better to have `getClientInfo(String clientId, String clientRole)` and make decisions based on the role

Reply:
Included role and name also is continued to make sure user name in test cases are intact